cairo/ethereum/cancun/fork.cairo

```diff
@@ -195,3 +221,322 @@ func make_receipt{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr: Kec
+    }
+    let sender_balance_after_gas_fee = sender_account.value.balance.value.low - effective_gas_fee -
+        blob_gas_fee.value;
+    with_attr error_message("OverflowError") {
+        assert [range_check_ptr] = sender_balance_after_gas_fee;
+        let range_check_ptr = range_check_ptr + 1;
+    }
+    tempvar sender_balance_after_gas_fee_u256 = U256(
+        new U256Struct(sender_balance_after_gas_fee, 0)
+    );
+    set_account_balance{state=state}(sender, sender_balance_after_gas_fee_u256);
+    EnvImpl.set_state{env=env}(state);
+
+    // Create preaccessed addresses and write coinbase
+    let (preaccessed_addresses_ptr) = dict_new_empty();
+    tempvar preaccessed_addresses_ptr_start = preaccessed_addresses_ptr;
+    let address = env.value.coinbase;
+    hashdict_write{poseidon_ptr=poseidon_ptr, dict_ptr=preaccessed_addresses_ptr}(
+        1, &address.value, 1
+    );
+    tempvar preaccessed_addresses = SetAddress(
+        new SetAddressStruct(
+            dict_ptr_start=cast(preaccessed_addresses_ptr_start, SetAddressDictAccess*),
+            dict_ptr=cast(preaccessed_addresses_ptr, SetAddressDictAccess*),
+        ),
+    );
+    // Create preaccessed storage keys
+    let (preaccessed_storage_keys_ptr) = dict_new_empty();
+    tempvar preaccessed_storage_keys = SetTupleAddressBytes32(
+        new SetTupleAddressBytes32Struct(
+            dict_ptr_start=cast(preaccessed_storage_keys_ptr, SetTupleAddressBytes32DictAccess*),
+            dict_ptr=cast(preaccessed_storage_keys_ptr, SetTupleAddressBytes32DictAccess*),
+        ),
+    );
+
+    if (tx.value.legacy_transaction.value == 0) {
+        let access_list = tx.value.access_list_transaction.value.access_list;
+        process_access_list{
+            poseidon_ptr=poseidon_ptr,
+            preaccessed_addresses=preaccessed_addresses,
+            preaccessed_storage_keys=preaccessed_storage_keys,
+        }(access_list, access_list.value.len, 0);
+        tempvar keccak_ptr = keccak_ptr;
+        tempvar bitwise_ptr = bitwise_ptr;
+        tempvar poseidon_ptr = poseidon_ptr;
+        tempvar preaccessed_addresses = preaccessed_addresses;
+        tempvar preaccessed_storage_keys = preaccessed_storage_keys;
+    } else {
+        tempvar keccak_ptr = keccak_ptr;
+        tempvar bitwise_ptr = bitwise_ptr;
+        tempvar poseidon_ptr = poseidon_ptr;
+        tempvar preaccessed_addresses = preaccessed_addresses;
+        tempvar preaccessed_storage_keys = preaccessed_storage_keys;
+    }
+
+    let keccak_ptr = cast([ap - 5], KeccakBuiltin*);
+    let bitwise_ptr = cast([ap - 4], BitwiseBuiltin*);
+    let poseidon_ptr = cast([ap - 3], PoseidonBuiltin*);
+    let preaccessed_addresses = SetAddress(cast([ap - 2], SetAddressStruct*));
+    let preaccessed_storage_keys = SetTupleAddressBytes32(
+        cast([ap - 1], SetTupleAddressBytes32Struct*)
+    );
+
+    tempvar code_address = OptionalAddress(cast(0, Address*));
```

Comment:
im surprised this compiles, given that the inner is supposed to be:

```suggestion
    tempvar code_address = OptionalAddress(cast(0, felt*));
```

I guess it's ok because it's a pointer of size 1